domain/test/validation.test.js

```diff
@@ -4,216 +4,316 @@ const validation = require('../src/validation');
-        assert(validationResult.item === undefined);
-        assert(validationResult.isValid === false);
-        assert(validationResult.errors.length === 1);
-    });
+    describe('for item', function () {
 
-    describe('validation error', function () {
-
-        it('#notValidType: [type]', function () {
+        it('предмет проходит валидацию', function () {
             let item = {
-                type: 'enot',
+                type: 'axe',
                 minDmg: 2,
                 maxDmg: 3,
                 critChance: 20,
                 critDmg: 60
             };
             let validationResult = validation.checkItem(item);
-            assert(validationResult.item === undefined);
-            assert(validationResult.isValid === false);
-            expect(validationResult.errors)
-                .to.have.deep.members([
-                    {
-                        id: "notValidType",
-                        properties: ["type"]
-                    }
-                ]);
+            assert(validationResult.item.type === item.type);
+            assert(validationResult.item.minDmg === item.minDmg);
+            assert(validationResult.item.maxDmg === item.maxDmg);
+            assert(validationResult.item.critChance === item.critChance);
+            assert(validationResult.item.critDmg === item.critDmg);
+            assert(validationResult.isValid === true);
+            assert(validationResult.errors.length === 0);
         });
 
-        it('#mustBeNumber: [minDmg]', function () {
+        it('избыточные свойства обрезаются и предмет проходит валидацию', function () {
             let item = {
                 type: 'axe',
-                minDmg: '2',
+                minDmg: 2,
                 maxDmg: 3,
                 critChance: 20,
-                critDmg: 60
+                critDmg: 60,
+                enot: true
             };
             let validationResult = validation.checkItem(item);
-            assert(validationResult.item === undefined);
-            assert(validationResult.isValid === false);
-            expect(validationResult.errors)
-                .to.have.deep.members([
-                    {
-                        id: "mustBeNumber",
-                        properties: ["minDmg"]
-                    }
-                ]);
+            assert(Object.keys(validationResult.item).length === Object.keys(item).length - 1);
+            assert(validationResult.item.type === item.type);
+            assert(validationResult.item.minDmg === item.minDmg);
+            assert(validationResult.item.maxDmg === item.maxDmg);
+            assert(validationResult.item.critChance === item.critChance);
+            assert(validationResult.item.critDmg === item.critDmg);
+            assert(validationResult.item.enot === undefined);
+            assert(validationResult.isValid === true);
+            assert(validationResult.errors.length === 0);
```

Comment:
Such asserts are not readable and hell of a pain to maintain and write.
See http://chaijs.com/api/assert/#method_deepequal
And something like
```js
assert.deepEqual(validationResult, {
  isValid: true,
  errors: [],
  item: {
    type = 'axe',
    ...
  }
});
```
is much more readable.

But, to be honest, it is not really required for assert for everything in every test. In this particular test you want to check that extra properties are skipped and that validation is passed (that's what the test title specifies). So you should assert ONLY for that:
```js
assert(validationResult.isValid === true);
assert(validationResult.item.enot === undefined);
```
If you want to test that validation preserves other properties and does not change them, you should have a separate test just for that. You don't want to enumerate all the properties in every test. The related topic is called "property testing", you can google about it for some more info.